### PR TITLE
Refactor machine inspect

### PR DESCRIPTION
**Pull request by baude**
I was asked to refactor machine inspect output to represent more common
and basic information.  machine inspect now has information that would
be appropriate for different machines.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
@n1hility we would need a similar change for windows ... is that something you could possibly crank out once this merges?

File: pkg/machine/config.go [hunk 1]
```diff
@@ -183,3 +193,121 @@ func GetConfDir(vmType string) (string, error) {
...
+
+// ResourceConfig describes physical attributes of the machine
+type ResourceConfig struct {
+	// CPUs to be assigned to the VM
+	CPUs uint64
+	// Memory in megabytes assigned to the vm
+	Memory uint64
+	// Disk size in gigabytes assigned to the vm
+	DiskSize uint64
+}
```
Comment (TomSweeneyRedHat) [inline pkg/machine/config.go:205]: To keep my OCD happy, could you flip the order of  Memory and DiskSize please?

File: pkg/machine/config.go [hunk 2]
```diff
@@ -183,3 +193,121 @@ func GetConfDir(vmType string) (string, error) {
...
+	m.Symlink = &sl
+	return os.Symlink(m.Path, sl)
+}
+
+type Mount struct {
+	Type     string
+	Tag      string
+	Source   string
+	Target   string
+	ReadOnly bool
```
Comment (TomSweeneyRedHat) [inline pkg/machine/config.go:283]: more OCD ordering

File: pkg/machine/config.go [hunk 3]
```diff
@@ -183,3 +193,121 @@ func GetConfDir(vmType string) (string, error) {
...
+// ImageConfig describes the bootable image for the VM
+type ImageConfig struct {
+	// IgnitionFile is the path to the filesystem where the
+	// ignition file was written (if needs one)
+	IgnitionFile VMFile `json:"IgnitionFilePath"`
+	// ImageStream is the update stream for the image
+	ImageStream string
+	// ImageFile is the fq path to
+	ImageFile VMFile `json:"ImagePath"`
+}
```
Comment (TomSweeneyRedHat) [inline pkg/machine/config.go:295]: OCD

File: pkg/machine/wsl/machine.go [hunk 4]
```diff
@@ -155,9 +155,9 @@ type Provider struct{}
...
 	// IdentityPath is the fq path to the ssh priv key
 	IdentityPath string
-	// IgnitionFilePath is the fq path to the .ign file
+	// IgnitionFile is the fq path to the .ign file
 	ImageStream string
-	// ImagePath is the fq path to
+	// ImageFile is the fq path to
 	ImagePath string
 	// Name of the vm
 	Name string
```
Comment (TomSweeneyRedHat) [inline pkg/machine/wsl/machine.go:163]: more OCD

**Review by n1hility — APPROVED** [review]
LGTM

**Comment by n1hility** [comment]
> @n1hility we would need a similar change for windows ... is that something you could possibly crank out once this merges?



> @n1hility we would need a similar change for windows ... is that something you could possibly crank out once this merges?

Sure will follow up immediately.

File: pkg/machine/config.go [hunk 5]
```diff
@@ -183,3 +193,121 @@ func GetConfDir(vmType string) (string, error) {
...
+
+// ImageConfig describes the bootable image for the VM
+type ImageConfig struct {
+	// IgnitionFile is the path to the filesystem where the
+	// ignition file was written (if needs one)
+	IgnitionFile VMFile `json:"IgnitionFilePath"`
+	// ImageStream is the update stream for the image
+	ImageStream string
+	// ImageFile is the fq path to
+	ImageFile VMFile `json:"ImagePath"`
```
Comment (Luap99) [inline pkg/machine/config.go:294]: Why do tag names not match the field names, that is confusing?

File: pkg/machine/wsl/machine.go [hunk 6]
```diff
@@ -155,9 +155,9 @@ type Provider struct{}
 type MachineVM struct {
 	// IdentityPath is the fq path to the ssh priv key
 	IdentityPath string
-	// IgnitionFilePath is the fq path to the .ign file
+	// IgnitionFile is the fq path to the .ign file
```
Comment (Luap99) [inline pkg/machine/wsl/machine.go:158]: Comment does not match field name, already before but please fix this

File: pkg/machine/wsl/machine.go [hunk 7]
```diff
@@ -155,9 +155,9 @@ type Provider struct{}
...
 	ImageStream string
-	// ImagePath is the fq path to
+	// ImageFile is the fq path to
```
Comment (Luap99) [inline pkg/machine/wsl/machine.go:160]: same here

**Pull request merged**